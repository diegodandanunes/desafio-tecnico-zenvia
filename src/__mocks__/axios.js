export default {
    get: jest.fn(() => Promise.resolve({data: [{char_id: 0, occupation: []}]}))
}