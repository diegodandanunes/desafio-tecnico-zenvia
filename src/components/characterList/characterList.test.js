import React from 'react';
import { render, cleanup, waitFor } from '@testing-library/react';

import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import CharacterList from './';
import fakeData from '../../__mocks__/character-list';

import axios from 'axios';

const history = createMemoryHistory();

afterEach(cleanup);

it('renders list', async () => {
    const { asFragment } = render(<Router history={history}><CharacterList /></Router>);
    await waitFor(() => expect(asFragment()).toMatchSnapshot());
});

it('display a list of characters', async () => {
    axios.get.mockImplementationOnce(() => Promise.resolve(fakeData));
    const { getByTestId } = render(<Router history={history}><CharacterList /></Router>);
    const list = await waitFor(() => getByTestId('character-list'));
    expect(list.children.length).toBe(fakeData.data.length);
})