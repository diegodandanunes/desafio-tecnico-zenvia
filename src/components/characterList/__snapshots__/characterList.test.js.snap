// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list 1`] = `
<DocumentFragment>
  <div
    class="sc-AxirZ iXyIpx"
  >
    <div
      class="min-h-100 display-flex"
    >
      <div
        class="display-flex flex-column w-100"
      >
        <header>
          <div
            class="sc-AxirZ iXyIpx"
          >
            <div
              class="sc-AxgMl kGAIUv"
            >
              <img
                alt="Logo Diego Nunes"
                src="logo.png"
              />
              <ul>
                <li>
                  Desafio técnico 
                  <span>
                    |
                  </span>
                   
                  <strong>
                    Zenvia
                  </strong>
                </li>
              </ul>
            </div>
          </div>
        </header>
        <div
          class="sc-AxhCb fkGitN display-flex align-center justify-center flex-grow flex-wrap"
        >
          <div
            class="sc-AxhUy jWPSrp"
          >
            <div
              class="lds-dual-ring"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders list 2`] = `
<DocumentFragment>
  <div
    class="sc-AxirZ iXyIpx"
  >
    <div
      class="min-h-100 display-flex"
    >
      <div
        class="display-flex flex-column w-100"
      >
        <header>
          <div
            class="sc-AxirZ iXyIpx"
          >
            <div
              class="sc-AxgMl kGAIUv"
            >
              <img
                alt="Logo Diego Nunes"
                src="logo.png"
              />
              <ul>
                <li>
                  Desafio técnico 
                  <span>
                    |
                  </span>
                   
                  <strong>
                    Zenvia
                  </strong>
                </li>
              </ul>
            </div>
          </div>
        </header>
        <div
          class="sc-AxhCb fkGitN display-flex align-center justify-center flex-grow flex-wrap"
        >
          <div
            class="display-flex flex-wrap justify-center"
            data-testid="character-list"
          >
            <div
              class="sc-AxheI bZrttE"
            >
              <a
                data-testid="card-link"
                href="/character/0"
              >
                <div
                  class="card-overlay"
                />
                <img
                  data-testid="card-image"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
